bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/ESHTrustManager.java

```diff
@@ -0,0 +1,187 @@
+        }
+        throw new IllegalStateException("Could not find X509ExtendedTrustManager");
+    }
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        checkClientTrusted(chain, authType, (Socket) null);
+    }
+
+    @Override
+    public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        checkServerTrusted(chain, authType, (Socket) null);
+    }
+
+    @Override
+    public X509Certificate[] getAcceptedIssuers() {
+        return defaultTrustManager.getAcceptedIssuers();
+    }
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType, Socket socket)
+            throws CertificateException {
+        X509ExtendedTrustManager linkedTrustManager = getLinkedTrustMananger(chain);
+        if (linkedTrustManager == null) {
+            logger.trace("No specific trust manager found, falling back to default");
+            defaultTrustManager.checkClientTrusted(chain, authType, socket);
+        } else {
+            linkedTrustManager.checkClientTrusted(chain, authType, socket);
+        }
+    }
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType, SSLEngine sslEngine)
+            throws CertificateException {
+        X509ExtendedTrustManager linkedTrustManager = getLinkedTrustMananger(chain);
+        if (linkedTrustManager == null) {
+            logger.trace("No specific trust manager found, falling back to default");
+            defaultTrustManager.checkClientTrusted(chain, authType, sslEngine);
+        } else {
+            linkedTrustManager.checkClientTrusted(chain, authType, sslEngine);
+        }
+    }
+
+    @Override
+    public void checkServerTrusted(X509Certificate[] chain, String authType, Socket socket)
+            throws CertificateException {
+        X509ExtendedTrustManager linkedTrustManager = getLinkedTrustMananger(chain);
+        if (linkedTrustManager == null) {
+            logger.trace("No specific trust manager found, falling back to default");
+            defaultTrustManager.checkServerTrusted(chain, authType, socket);
+        } else {
+            linkedTrustManager.checkServerTrusted(chain, authType, socket);
+        }
+    }
+
+    @Override
+    public void checkServerTrusted(X509Certificate[] chain, String authType, SSLEngine sslEngine)
+            throws CertificateException {
+        X509ExtendedTrustManager linkedTrustManager = getLinkedTrustMananger(chain);
+        if (linkedTrustManager == null) {
+            logger.trace("No specific trust manager found, falling back to default");
+            defaultTrustManager.checkServerTrusted(chain, authType, sslEngine);
+        } else {
+            linkedTrustManager.checkServerTrusted(chain, authType, sslEngine);
+        }
+    }
+
+    private X509ExtendedTrustManager getLinkedTrustMananger(X509Certificate[] chain) {
```

Comment:
Note to self or someone that knows, I found another implementation which did not assume this chain to be ordered, this should of course be checked